apiserver/src/main/java/org/dependencytrack/resources/v1/IntegrationResource.java

```diff
@@ -94,4 +103,34 @@ public Response getInactiveEcosystems() {
                 .collect(Collectors.toList());
         return Response.ok(ecosystems).build();
     }
+
+    @POST
+    @Path("gitlab/{state}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Set state of gitlab integration", description = "<p>Requires permission <strong>SYSTEM_CONFIGURATION</strong> or <strong>SYSTEM_CONFIGURATION_CREATE</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "GitLab state set successfully"),
+            @ApiResponse(responseCode = "304", description = "The GitLab integration is already in the desired state"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized")
+    })
+    @PermissionRequired({ Permissions.Constants.SYSTEM_CONFIGURATION,
+            Permissions.Constants.SYSTEM_CONFIGURATION_CREATE }) // Require admin privileges due to system impact
+    public Response handleGitLabStateChange(
+            @Parameter(description = "A valid boolean", required = true) @PathParam("state") String state) {
+        try (QueryManager qm = new QueryManager()) {
+            final ConfigProperty property = qm.getConfigProperty(GITLAB_ENABLED.getGroupName(),
+                    GITLAB_ENABLED.getPropertyName());
+
+            if (!property.getPropertyValue().equals(state)) {
+                if (!state.equalsIgnoreCase("true") && !state.equalsIgnoreCase("false")) {
+                    return Response.status(Response.Status.BAD_REQUEST).build();
+                }
+                property.setPropertyValue(state);
+                qm.persist(property);
```

Comment:
Please wrap this in a transaction using `qm.callInTransaction`.

Note that `Event.dispatch` must happen *outside* of the transaction. If the TRX has to be rolled back for some reason we don't want the event to be dispatched.